HW/baseball.sql

```diff
@@ -0,0 +1,29 @@
+
+-- first join the game and batter counts table and save the results as a temporary table
+drop temporary table if exists joined_batter_game;
+create temporary table joined_batter_game
+select game.local_date,batter_counts.batter,batter_counts.game_id,batter_counts.Hit,batter_counts.atBat from batter_counts join game on batter_counts.game_id=game.game_id;
+-- change batter data type to character because it was integer
+alter table joined_batter_game modify batter varchar(100);
+-- calculate annual batting average
+drop temporary table if exists annual_at_bat_per_player;
+create temporary table annual_at_bat_per_player
+select batter,year(local_date), sum(Hit)/sum(nullif(atBat,0)) as Average_at_bat from joined_batter_game group by batter,year(local_date) order by year(local_date);
+-- view the table annual_at_bat_per_player
+select * from annual_at_bat_per_player order by batter limit 1,20;
+-- calculate historic batting average
+drop temporary table if exists historic_at_bat_per_player;
+create temporary table historic_at_bat_per_player
+select batter, sum(Hit)/sum(nullif(atBat,0)) as Average_at_bat from joined_batter_game group by batter;
+-- view the table historic_at_bat_per_player
+select * from historic_at_bat_per_player order by batter limit 1,20;
+-- calculate last 100 days batting average, first we create a table with the 100 recent entries for each player
+drop temporary table if exists last_100_days_per_player;
+create temporary table last_100_days_per_player
+select * from (select * , row_number() over (partition by batter order by local_date desc) as date_rank from joined_batter_game) ranks where date_rank <=99;
+-- Then we calculate the batting average for each player
+drop temporary table if exists last_100_days_batting_per_player;
+create temporary table last_100_days_batting_per_player
+select batter, sum(Hit)/sum(nullif(atBat,0)) as batting_average from last_100_days_per_player group by batter;
```

Comment:
So this is the one to change. We're looking for a 100 day average for each batter, for each day that they played. So if you run it for a single batter, you would expect one row for each date that they have a game, minus any games where they didn't play for the 100 days prior (like the start of a season).

First, try setting up a temporary table with just a few thousand entries. Maybe for a single batter.
Then try joining the table on itself. so: 
select * 
from table b1 
left join table b2 on b1.batter = b2.batter;
You'll have a huge table now.
Then add a 2 part group by, because we want to group by batter (once you run it for more than one batter), and group by date.
Next switch what you're selecting. Instead of selecting a number of hits or at bats, select a sum with a "case when" the date from the b1 table is less than 100 days from the date in the b2 table. I used datediff and between for this, but someone in class mentioned using a date add function or a date subtract function.

Last thing to get all the points, make sure you only sample games in the past for this average. Don't include the current game. Otherwise, if you use this data to train a model, it would be cheating.